packages/config/src/project/index.ts

```diff
@@ -0,0 +1,134 @@
+import _ from 'lodash'
+import Debug from 'debug'
+import deepDiff from 'return-deep-diff'
+
+import errors, { ConfigValidationFailureInfo, CypressError } from '@packages/errors'
+import { getCtx } from '@packages/data-context/src/globalContext'
```

Comment:
This dependency cannot be removed until we move [`getFilesByGlob `](https://github.com/cypress-io/cypress/blob/1b688694fe7ad7de7ce4ac215024cd128d68bf39/packages/data-context/src/sources/FileDataSource.ts#L31-L57) to somewhere else.

Or call it inside `data-context` and pass the value to this function. 

Maybe, this can be solved in the future PRs.

Reply:
I researched it after writing this comment. It seems that it will work if we add `supportFiles` member to `setupFullConfigWithDefaults`.

I'll do this after this PR is merged. This PR is big and I tried not to change code much.